server/controllers/finance/cash.js

```diff
@@ -0,0 +1,247 @@
+* currencies.  The API accepts a cashbox ID during cash payment creation and
+* looks up the correct account based on the cashbox_id + currency.
+*/
+var db   = require('../../lib/db');
+var uuid = require('../../lib/guid');
+
+// check if an object is empty
+// TODO - can this be from a library?
+function empty(object) {
+  return Object.keys(object).length === 0;
+}
+
+
+// looks up a single cash record and associated cash_items
+// sets the "canceled" flag if a cash_discard record exists.
+function lookupCashRecord(uuid, codes) {
+  'use strict';
+
+  var record;
+  var cashRecordSql =
+    'SELECT cash.uuid, CONCAT(project.abbr, cash.reference) AS reference, ' +
+      'cash.date, cash.debtor_uuid, cash.currency_id, cash.amount, ' +
+      'cash.description, cash.cashbox_id, cash.is_caution, cash.user_id ' +
+    'FROM cash JOIN project ON cash.project_id = project.id ' +
+    'WHERE cash.uuid = ?;';
+
+  var cashItemsRecordSql =
+    'SELECT cash_item.uuid, cash_item.amount, cash_item.invoice_uuid ' +
+    'FROM cash_item WHERE cash_item.cash_uuid = ?;';
+
+  var cashDiscardRecordSql =
+    'SELECT cash_uuid FROM cash_discard WHERE cash_uuid = ?;';
+
+  return db.exec(cashRecordSql, [uuid])
+  .then(function (rows) {
+
+    if (rows.length === 0) {
+      throw codes.ERR_NOT_FOUND;
+    }
+
+    // store the record for return
+    record = rows[0];
+
+    return db.exec(cashItemsRecordSql, [uuid]);
+  })
+  .then(function (rows) {
+
+    // bind the cash items to the "items" property and return
+    record.items = rows;
+
+    return db.exec(cashDiscardRecordSql, [uuid]);
+  })
+  .then(function (rows) {
+
+    // if a linked cash_discard record exists, it means that this cash record
+    // has been reversed and we'll report that using a 'canceled' flag.
+    record.canceled = rows.length > 0;
+
+    return record;
+  });
+}
+
+/**
+ * GET /cash
+ * TODO - Query Paramters:
+ *   start={date}
+ *   end={date}
+ *   debtor={uuid}
+ *   type={ 'cash' | 'caution' },
+ *   project={id}
+ *
+ * Lists the cash payments with optional filtering parameters.
+ * @returns payments An array of { uuid, reference, date } JSON objects
+ */
+exports.list = function list(req, res, next) {
+  'use strict';
+
+  // base query
+  var sql =
+    'SELECT cash.uuid, CONCAT(project.abbr, cash.reference) AS reference, ' +
+      'cash.date, cash.amount '  +
+    'FROM cash JOIN project ON cash.project_id = project.id;';
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+ * GET /cash/:uuid
+ *
+ * Get the details of a particular cash payment.  This endpoint will return a
+ * record in the following format:
+ * {
+ *   uuid : "..",
+ *   items : [ {}, {} ]    // items hitting invoices, if applicable
+ *   is_caution : true/false,
+ *   canceled: true/false, // indicates if a cash_discard record exists
+ *   ...
+ *  }
+ */
+exports.details = function details(req, res, next) {
```

Comment:
Since this function will only ever be called from an HTTP "GET" request, the "get" in "getDetails" is implied.  Here's what it looks like from the server:
```js
app.get('/cash/:uuid', cash.getDetails);
// versus
app.get('/cash/:uuid', cash.details);
```
In the first call, get is repeated twice, in the second one it reads cleanly.